Skeletonize

jsf9k opened the pull request:
## 🗣 Description ##

This pull request skeletonizes this repository and brings it into line with the analogous changes that were made to [cisagov/trustymail_reporter](https://github.com/cisagov/trustymail_reporter).

## 💭 Motivation and context ##

This repository deserves to get Lineage updates too!

This pull request also resolves #8.

## 🧪 Testing ##

I built a new Docker image with these changes and successfully used it to perform a full run of the BOD 18-01 scanning.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] _All_ future TODOs are captured in issues, which are referenced in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Post-merge checklist ##

- [ ] Add a tag.

mcdonnnj committed:
Merge pull request #55 from cisagov/migrate_prettier_hook

Change repo for the prettier pre-commit hook

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into lineage/skeleton

jsf9k committed:
Merge pull request #37 from cisagov/lineage/skeleton

⚠️ CONFLICT! Lineage pull request for: skeleton

felddy committed:
Add script to generate cross-platform Dockerfile.

felddy committed:
Add git ignore for generated Dockerfile-x.

felddy committed:
Replace previous Docker workflows with a moderinzed, complete-workflow.

These changes are based on the "complete workflow" from
https://github.com/docker/build-push-action

Additionally it adds support for the new "workflow_dispatch" event type.

felddy committed:
Add CodeQL workflow. 

Needed to check tests written in Python.

felddy committed:
Update README publisher to use new Docker URL.

felddy committed:
Expand command option names to long form.

felddy committed:
Update Dockerfile to use opencontainers metadata and build-time labels.

felddy committed:
Prevent image pushes when linter checks fail.

felddy committed:
Modify cache keys to follow team-established naming conventions.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

felddy committed:
Add comments pointing to the opencontainers image-spec documentation.

felddy committed:
Add comments to help document and break up a monolitic file.

jsf9k committed:
Insist that the cisagov devs are the owners of the .github directory

This additional clause must remain at the _end_ of the CODEOWNERS file
so that it cannot be overridden by a later clause.

We want to make it so that all the .github files including CODEOWNERS
are protected so only code owners (the dev team) can approve
modifications to them.

This will prevent configuration changes from breaking Actions and
other management-type functions that the files in this directory
control. By setting the .github files/folder to require code owner
approval for changes, workflow and management changes will require dev
team review and checking.

Resolves #56.

jsf9k committed:
Remove offending slash

Thanks to @dav3r and @mcdonnnj for the suggestion.

felddy committed:
Apply correct typographic designation to character replacement comment.

felddy committed:
Modernize Docker shields in README to point to working links.

felddy committed:
Elaborate on the Solidus.

Add some additional comments in case anyone is confused about the nature of the solidus and its place within our tagging system.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

felddy committed:
Merge pull request #38 from cisagov/improvement/modern_docker_workflow

Modernize Docker workflow

jsf9k committed:
Use the python version output by actions/setup-python

There is no need to run python code to determine the python version.
Resolves #58.

See here for details:
https://github.com/actions/setup-python/blob/main/action.yml#L14-L16

mcdonnnj committed:
Organize pre-commit hooks

Group related pre-commit hooks together. Make sure that hooks are
alphabetically sorted within those groups.

mcdonnnj committed:
Merge pull request #60 from cisagov/organize_pre-commit_requirements

Organize and Sort pre-commit Hooks

jsf9k committed:
Merge branch 'develop' into improvement/insist-that-devs-own-gihub-dir

jsf9k committed:
Merge pull request #57 from cisagov/improvement/insist-that-devs-own-gihub-dir

Ensure that the cisagov devs are the owners of the .github directory

jsf9k committed:
Merge branch 'develop' into improvement/get-python-version-more-simply

jsf9k committed:
Merge pull request #59 from cisagov/improvement/get-python-version-more-simply

Use the python version output by actions/setup-python

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into lineage/skeleton

mcdonnnj committed:
Fix test job missing Python version in cache key

mcdonnnj committed:
Merge pull request #40 from cisagov/bugfix/fix_incorrect_caching

Add Python Version Missing from actions/cache Key

mcdonnnj committed:
Merge branch 'develop' into lineage/skeleton

mcdonnnj committed:
Merge pull request #39 from cisagov/lineage/skeleton

⚠️ CONFLICT! Lineage pull request for: skeleton

mcdonnnj committed:
Clean up our actions/cache step

Removed name because it was not more informative than the default. Swapped out
a hardcoded job reference for the github.job context value. Switch the base
cache key to a step environment value so we can set it once and reuse. Removed
additional restore-key value that might have undesirable results.

mcdonnnj committed:
Merge pull request #62 from cisagov/improvement/clean_up_actions/cache

Improve actions/cache Configuration

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Update remaining actions/cache uses

Make sure all actions/cache steps are in-line with the changes made to the lint
job's step.

mcdonnnj committed:
Merge pull request #41 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Update prettier hook details

Per https://github.com/prettier/prettier/pull/8937 the pre-commit hook has
been moved to https://github.com/pre-commit/mirrors-prettier. I have also
updated to the latest version in that repository.

mcdonnnj committed:
Update black hook repo url

At some point the python GitHub organization renamed to psf (Python Software
Foundation). Although it redirects with no issue, I think it should be updated
to reflect the current repository URL.

mcdonnnj committed:
Remove trailing '.git's from repository URLs

Two hooks had trailing '.git's in the URLs. Although this is not a problem, we
should be consistent in how we format things.

mcdonnnj committed:
Enable two additional hooks from default pre-commit-hooks

Enabled 'check-case-conflict' because of our mixed Linux and macOS development.
Although APFS supports case-sensitive containers, it is not the default as far
as I am aware. Linux filesystems are typically case-sensitive however.

With the merge of https://github.com/cisagov/development-guide/pull/42 we now
have a TOML file in a repository, so it does not hurt to add this hook in case
more are added in the future.

mcdonnnj committed:
Run pre-commit autoupdate

mcdonnnj committed:
Update repo URL for the isort hook

I missed that this repository was transfered from the creator, Timothy Crosley,
to the PyCQA organization.

mcdonnnj committed:
Update ansible-lint hook URL

mcdonnnj committed:
Run pre-commit autoupdate

mcdonnnj committed:
Add the pre-commit-packer repo and hooks

mcdonnnj committed:
Add the --strict flag to the yamllint pre-commot hook

The --strict flag will cause the yamllint hook to exit with a non-zero exit
code when warnings are found instead of only when errors are found.

mcdonnnj committed:
Merge pull request #63 from cisagov/pre-commit_hook_maintenance

pre-commit Hook Maintenance and Update

mcdonnnj committed:
Merge branch 'develop' into improvement/add_pre-commit-packer

mcdonnnj committed:
Merge pull request #64 from cisagov/improvement/add_pre-commit-packer

Add the pre-commit-packer Repo and Hooks to the pre-commit Configuration

mcdonnnj committed:
Merge branch 'develop' into improvement/enable_yamllint_strict

mcdonnnj committed:
Merge pull request #65 from cisagov/improvement/enable_yamllint_strict

Use the --strict Flag for the yamllint pre-commit Hook

mcdonnnj committed:
Merge github.com:cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Merge pull request #43 from cisagov/lineage/skeleton

⚠️ CONFLICT! Lineage pull request for: skeleton

mcdonnnj committed:
Add the beautysh pre-commit hook

This hook performs auto-formatting (beautifying) of Bash scripts.

mcdonnnj committed:
Update pre-commit hooks with `pre-commit autoupdate`

mcdonnnj committed:
Add a requirements-dev.txt pip requirements file

Our other skeletons have this file, even if they are not Python projects. The
directions in CONTRIBUTING.md reference such a file, which can be misleading.
Rather than amending the CONTRIBUTING documentation, I feel it's better to
align with our other skeletons for consistency.

mcdonnnj committed:
Revert ansible-lint version update

The v5 release of ansible-lint introduces breaking changes, so we are holding
off on updating until things are resolved. Conversation about this can be
tracked in https://github.com/cisagov/skeleton-ansible-role/issues/69.

mcdonnnj committed:
Merge pull request #69 from cisagov/maintenance/update_pre-commit_hooks

Update pre-commit Hooks

mcdonnnj committed:
Merge branch 'develop' into improvement/add_beautysh_pre-commit_hook

mcdonnnj committed:
Merge pull request #67 from cisagov/improvement/add_beautysh_pre-commit_hook

Add the beautysh pre-commit hook

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Autoformat bump_version.sh with beautysh

mcdonnnj committed:
Autoformat push_readme.sh with beautysh

mcdonnnj committed:
Fix the agency name in the vendor label

mcdonnnj committed:
Merge pull request #46 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into improvement/fix_agency_name

mcdonnnj committed:
Merge pull request #47 from cisagov/improvement/fix_agency_name

Correct the Agency Name for Docker Images

mcdonnnj committed:
Merge branch 'develop' into bug/add_requirements-dev_file

mcdonnnj committed:
Merge pull request #45 from cisagov/bug/add_requirements-dev_file

Add a requirements-dev.txt pip Requirements File

mcdonnnj committed:
Update CodeQL workflow for GitHub Actions

I noticed the following warning when looking at GHA workflow runs:

1 issue was detected with this workflow: git checkout HEAD^2 is no longer
necessary. Please remove this step as Code Scanning recommends analyzing the
merge commit for best results.

After looking into it I updated the workflow per this page on GitHub Docs:

https://docs.github.com/en/github/finding-security-vulnerabilities-and-errors-in-your-code/troubleshooting-the-codeql-workflow#warning-git-checkout-head2-is-no-longer-necessary

mcdonnnj committed:
Merge pull request #48 from cisagov/bugfix/update_codeql_workflow

Update CodeQL Workflow for GitHub Actions

felddy committed:
Enable container publication to GitHub Container Registry.

* Duplicates DockerHub tags for ghcr.io
* Adds login to GitHub Container Registry using the GITHUB_TOKEN secret.
* Documentation update to reflect multi-registry publication.

felddy committed:
Update .github/workflows/build.yml

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

felddy committed:
Add better build instructions to the README.

* Add instructions for cross-platform builds
* Clean up shields
* Clean up tables for each Docker configuration item
* Add section about supported tags

mcdonnnj committed:
Skip CodeQL workflow for Dependabot branches on push events

This should resolve the following error:

Error: Workflows triggered by Dependabot on the "push" event run with read-only
access. Uploading Code Scanning results requires write access. To use Code
Scanning with Dependabot, please ensure you are using the "pull_request" event
for this workflow and avoid triggering on the "push" event for Dependabot
branches. See
https://docs.github.com/en/code-security/secure-coding/configuring-code-scanning#scanning-on-push
for more information on how to configure these events.

felddy committed:
Update README.md

Co-authored-by: Hillary <hillary.jeffrey@trio.dhs.gov>

felddy committed:
Update README.md

Co-authored-by: Hillary <hillary.jeffrey@trio.dhs.gov>

felddy committed:
Update README.md

Co-authored-by: Hillary <hillary.jeffrey@trio.dhs.gov>

felddy committed:
Update README.md

Co-authored-by: Hillary <hillary.jeffrey@trio.dhs.gov>

felddy committed:
Update README.md

Co-authored-by: Hillary <hillary.jeffrey@trio.dhs.gov>

felddy committed:
Update README.md

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

felddy committed:
Update README.md

Co-authored-by: Hillary <hillary.jeffrey@trio.dhs.gov>

felddy committed:
Update README.md

Co-authored-by: Hillary <hillary.jeffrey@trio.dhs.gov>

felddy committed:
Update README.md

Co-authored-by: Hillary <hillary.jeffrey@trio.dhs.gov>

felddy committed:
Improve gitignore file comments and organization.

* Add a comment describing what the files does.
* Add sections to organize patterns.

mcdonnnj committed:
Update pre-commit hooks

Hooks updated with the `pre-commit autoupdate` command.

Note:
The `ansible-lint` hook is intentionally being held back on 4.3.7 because of
ongoing issues with the 5.x version and how we use Ansible (standalone Galaxy
roles).

mcdonnnj committed:
Remove user from CODEOWNERS

Due to her departure this removes hillaryj from the default CODEOWNERS we use
in our projects.

mcdonnnj committed:
Add semver package to development requirements

The semver package is needed by the bump_version.sh script, but it is not
included in requirements-dev.txt.

felddy committed:
Unquote port integers as this is now a parse error in Docker.

felddy committed:
Change compose specification version to match pyenv testing composition.

The version attribute has been depricated and will be removed once our 
systems move to a more recent release.

felddy committed:
Add context around command-line instructions for non-technical folks.

felddy committed:
Update README.md

Co-authored-by: Hillary <hillary.jeffrey@trio.dhs.gov>

mcdonnnj committed:
Add the tag.sh script to this repository

Since the GitHub Actions configuration builds images on tag pushes, we should
use the standard tagging script we use for the same purpose in other
repositories. This will make it easier to tag, build, and push Docker images
of a specific version in a standardized manner.

felddy committed:
Update README.md

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Update configuration instructions for pyenv

With the release of pyenv v2.0.0 there is a breaking change around startup
logic that necessitates an update for our pyenv setup instructions. Also
add a statement about how to get configuration instructions from pyenv
itself.

mcdonnnj committed:
Merge pull request #71 from cisagov/maintenance/update_pre-commit_hooks

Update pre-commit Hooks

mcdonnnj committed:
Merge branch 'develop' into maintenance/remove_user_from_codeowners

mcdonnnj committed:
Merge pull request #72 from cisagov/maintenance/remove_user_from_codeowners

Remove User from CODEOWNERS

mcdonnnj committed:
Merge branch 'develop' into documentation/update_pyenv_instructions

mcdonnnj committed:
Merge pull request #75 from cisagov/documentation/update_pyenv_instructions

Update Instructions for Configuring pyenv

mcdonnnj committed:
Merge branch 'develop' into improvement/gitignore_me

mcdonnnj committed:
Merge pull request #70 from cisagov/improvement/gitignore_me

Improve gitignore file comments and organization.

mcdonnnj committed:
Merge github.com:cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Add Docker section to the gitignore file

Mirror changes in cisagov/skeleton-generic by adding a Docker section to the
gitignore file and moving appropriate entries to that section.

mcdonnnj committed:
Merge pull request #59 from cisagov/lineage/skeleton

⚠️ CONFLICT! Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into improvement/skip_codeql_on_push_for_dependabot_branches

mcdonnnj committed:
Merge pull request #53 from cisagov/improvement/skip_codeql_on_push_for_dependabot_branches

Skip CodeQL Workflow for Dependabot Branches on push Events

mcdonnnj committed:
Merge branch 'develop' into bug/add_semver_to_requirements

mcdonnnj committed:
Merge pull request #55 from cisagov/bug/add_semver_to_requirements

Add semver to Development Requirements File

mcdonnnj committed:
Merge branch 'develop' into improvement/add_tag_script

mcdonnnj committed:
Merge pull request #57 from cisagov/improvement/add_tag_script

Add the tag.sh Script

mcdonnnj committed:
Merge branch 'develop' into improvement/gh_registry

mcdonnnj committed:
Merge pull request #49 from cisagov/improvement/gh_registry

Enable container publication to GitHub Container Registry.

mcdonnnj committed:
Merge branch 'develop' into improvement/readme_build_instructions

mcdonnnj committed:
Merge pull request #51 from cisagov/improvement/readme_build_instructions

Add better build instructions to the README.

mcdonnnj committed:
Standardize the layout of the Lineage configuration file

jsf9k committed:
Add a section to the README for ports

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Update the README to reference the latest version of the Docker container

This change was discussed among @cisagov/team-ois here:
https://github.com/cisagov/gatherer/pull/60#pullrequestreview-683086244

Also correct git comments to use the imperative mood.

jsf9k committed:
Merge pull request #62 from cisagov/improvement/use-latest-version-number-in-readme

Update the README to reference the latest version of the Docker container

jsf9k committed:
Document the single exposed (but non-functional) port

Co-authored-by: Nick M <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Merge pull request #61 from cisagov/documentation/add-ports-section

Add a section to the README for ports

mcdonnnj committed:
Add style enforcement rules

Add rules to enforce ATX-closed headers, dashes for unordered list elements,
and `1.` for ordered list elements.

mcdonnnj committed:
Add rule for image headers

Add <h1> and <img> tags to the allowed list for MD033 (HTML elements) to
support using an image as the first thing in a markdown file (header image).

mcdonnnj committed:
Switch to a YAML markdownlint configuration file

This converts the existing `.mdl_config.json` file to an equivalent
`.mdl_config.yaml` file. The reference in the markdownlint pre-commit hook
configuration is updated to match.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Add comments to markdownlint configuration

Now that this is a YAML file we can add comments explaining the rule
modifications we use. This will make it easier to edit or expand in the future.

mcdonnnj committed:
Bump actions/github-script pin from v3 to v4

mcdonnnj committed:
Add the validate_manifest hook from pre-commit

This hook will validate any pre-commit hook manifest files in the repository.

dependabot[bot] committed:
Bump python from 3.9-alpine to 3.9.6-alpine

Bumps python from 3.9-alpine to 3.9.6-alpine.

---
updated-dependencies:
- dependency-name: python
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

jsf9k committed:
Install terraform and packer for the linting job

We should be doing this because the Packer and Terraform pre-commit
hooks leverage the corresponding executables; therefore, it makes
sense to go ahead and install the particular versions of those
executables that we support.  Also add support for optionally
debugging via tmate.

See also cisagov/skeleton-generic#74.

jsf9k committed:
Break out the curl cache creation into its own step

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Remove unnecessary line in tasks

There is no reason to create /usr/bin/terraform.  This is a vestige of
an earlier age.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Modify the Packer installation to model that of Terraform

The Terraform installation does not destroy the existing system
Terraform installation, and neither should the Packer installation.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Install a specific version of terraform-docs

Note that this change is dependent on the merging of
cisagov/setup-env-github-action#31.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Move go installation so that it takes place before the cache task

Some variables defined in the go installation are used in the cache
task, so the go installation must happen first.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Capitalize Go for consistency

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

mcdonnnj committed:
Update pre-commit hooks

This is performed by running `pre-commit autoupdate`, but with the
`ansible-lint` hook held back manually.

jsf9k committed:
Prefer the newer "go install" syntax

As of [Go 1.16](https://tip.golang.org/doc/go1.16#go-command) the `GO111MODULE` environment variable defaults to `on` and `go get` has been deprecated for module installation.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

mcdonnnj committed:
Install the shfmt tool for GHA

The `shfmt` tool does not ship on the GitHub Actions runners so we must install
it manually.

mcdonnnj committed:
Replace the beautysh hook with pre-commit-shfmt

We have had a difficult time with how beautysh parses some shellscripts. I went
in pursuit of an alternative and I believe shfmt to be a good alternative.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Apply changes from the shfmt pre-commit hook

dav3r committed:
Use preferred capitalization for Docker Hub

dav3r committed:
Satisfy linter by shortening a long line

dav3r committed:
Use preferred capitalization for Docker Hub

dav3r committed:
Capitalize Docker in README

dav3r committed:
Fix additional Docker and Docker Hub capitalizations

mcdonnnj committed:
Merge pull request #83 from cisagov/maintenance/update_pre-commit_hooks

Update pre-commit Hooks

mcdonnnj committed:
Merge branch 'develop' into improvement/add_pre-commit_hook

mcdonnnj committed:
Merge pull request #80 from cisagov/improvement/add_pre-commit_hook

Add the validate_manifest Hook to pre-commit Configuration

mcdonnnj committed:
Merge branch 'develop' into improvement/update_mdl_configuration

jsf9k committed:
Merge branch 'develop' into improvement/install-tf-and-packer-for-linting

mcdonnnj committed:
Merge pull request #79 from cisagov/improvement/update_mdl_configuration

Update markdownlint Configuration

mcdonnnj committed:
Merge branch 'develop' into improvement/install-tf-and-packer-for-linting

mcdonnnj committed:
Merge pull request #82 from cisagov/improvement/install-tf-and-packer-for-linting

Install terraform and packer for the linting job

mcdonnnj committed:
Merge branch 'develop' into improvement/replace_beautysh_with_shfmt

mcdonnnj committed:
Merge pull request #84 from cisagov/improvement/replace_beautysh_with_shfmt

Replace beautysh hook with cisagov/pre-commit-shfmt

mcdonnnj committed:
Merge github.com:cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Satisfy pre-commit hooks

Update to reflect the new pre-commit configuration.

mcdonnnj committed:
Merge pull request #66 from cisagov/lineage/skeleton

⚠️ CONFLICT! Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into improvement/standardize_lineage_configuration

mcdonnnj committed:
Merge pull request #60 from cisagov/improvement/standardize_lineage_configuration

Standardize the Layout of the Lineage Configuration File

mcdonnnj committed:
Merge branch 'develop' into update/bump_github-script_to_v4

mcdonnnj committed:
Merge pull request #63 from cisagov/update/bump_github-script_to_v4

Update the Pin for actions/github-script to v4

mcdonnnj committed:
Merge branch 'develop' into dav3r-patch-1

jsf9k committed:
Merge branch 'develop' into dependabot/docker/python-3.9.6-alpine

mcdonnnj committed:
Merge pull request #65 from cisagov/dav3r-patch-1

Use preferred capitalization for Docker and Docker Hub

jsf9k committed:
Merge branch 'develop' into dependabot/docker/python-3.9.6-alpine

dav3r committed:
Fix a typo

jsf9k committed:
Remove support for the linux/s390x platform

Issue cisagov/skeleton-docker#69 has been created to remind us to
re-add it once it is again supported by the official Python Docker
image.

mcdonnnj committed:
Use the hashicorp/setup-terraform Action

Instead of manually installing a Terraform binary we can leverage the Action
provided by Hashicorp to do the same thing.

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`. The `ansible-lint` hook
is intentionally held back due to issues with switching to v5.

mcdonnnj committed:
Add a markdownlint rule for horizontal rules

This sets a specific rule for MD035 (Horizontal rule style) instead of the
default value of "consistent".

mcdonnnj committed:
Add a markdownlint rule for code blocks

This sets a specific rule for MD046 (code block style) instead of the default
value of "consistent".

mcdonnnj committed:
Merge branch 'develop' into dav3r-patch-1

mcdonnnj committed:
Merge pull request #85 from cisagov/dav3r-patch-1

Fix a typo

mcdonnnj committed:
Merge branch 'develop' into improvement/specify_mdl_rules

mcdonnnj committed:
Merge pull request #89 from cisagov/improvement/specify_mdl_rules

Add markdownlint Rules to Enforce Specific Formatting

mcdonnnj committed:
Merge branch 'develop' into improvement/use_setup-terraform_in_gha_workflow

mcdonnnj committed:
Merge pull request #88 from cisagov/improvement/use_setup-terraform_in_gha_workflow

Use the hashicorp/setup-terraform Action to Install Terraform in GHA Workflow

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

jsf9k committed:
Revert "Remove support for the linux/s390x platform"

This reverts commit 3ebf37da6ea14dd4bbdb607271d8faa9e1484dd2.

There was finally a successful build of the official Python Docker
image upstream, so we can add this back in.

Resolves cisagov/skeleton-docker#69.

dav3r committed:
Enable terraform validate pre-commit hook

This hook should now work as expected in most, if not all of our repos 
now that we are finally updating to terraform 0.13.x (on our way to 
1.0.x).

dav3r committed:
Temporarily use cisagov/setup-env-github-action@improvement/support_tf_0.13

This change will be reverted when testing is completed.

mcdonnnj committed:
Merge pull request #70 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into dependabot/docker/python-3.9.6-alpine

mcdonnnj committed:
Merge pull request #64 from cisagov/dependabot/docker/python-3.9.6-alpine

Bump python from 3.9-alpine to 3.9.6-alpine

dav3r committed:
Remove lint job step to initialize Terraform directories

Initialization will now be done during the "terraform validate" step.

dav3r committed:
Revert "Temporarily use cisagov/setup-env-github-action@improvement/support_tf_0.13"

This reverts commit 6a7fbf07bd371d0493c523ce24647e5c04c77c03.

dav3r committed:
Merge pull request #90 from cisagov/improvement/enable_terraform_validate

Enable terraform validate pre-commit hook

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

dependabot[bot] committed:
Bump actions/github-script from 4 to 5

Bumps [actions/github-script](https://github.com/actions/github-script) from 4 to 5.
- [Release notes](https://github.com/actions/github-script/releases)
- [Commits](https://github.com/actions/github-script/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/github-script
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`. The `ansible-lint` hook
is intentionally held back due to issues with switching to v5.

mcdonnnj committed:
Use an id when using cisagov/setup-env-github-action

mcdonnnj committed:
Use setup-env outputs for Terraform version

Change from using an environment variable to the outputs of the
cisagov/setup-env-github-action instead.

mcdonnnj committed:
Use setup-env outputs for Packer version

We use a `PACKER_VERSION` environment variable for the "Install Packer" step
that is populated from the `setup-env` outputs to get around `yamllint` lint
length limits.

mcdonnnj committed:
Use setup-env outputs for shfmt version

We change the "Install shfmt" step to use two local environment variables to
provide the package's URL and version to install. This allows us to work around
`yamllint` line length limits.

mcdonnnj committed:
Use setup-env outputs for Terraform-docs version

We change the "Install Terraform-docs" step to use two local environment
variables to provide the package's URL and version to install. This allows us
to work around `yamllint` line length limits.

mcdonnnj committed:
Store Go version as a step output

This changes from using an environment variable to using a step output to store
the Go version that is installed. This mirrors changes made to the other
program versions and how they're stored.

mcdonnnj committed:
Update the ansible-lint version for pre-commit

mcdonnnj committed:
Add an ansible-lint configuration file

This file is copied from cisagov/skeleton-ansible-role#85 and tweaked to pass
pre-commit. See here for for a list of the elements that can exist in this
file: https://ansible-lint.readthedocs.io/en/latest/configuring.html

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Add dependabot configuration for Terraform

This will configure `dependabot` to scan Terraform configurations if they exist
in a repository. If a repository stores a Terraform configuration in a sub-
directory this configuration will need to be modified or an additional
configuration must be added if there still exists a Terraform configuration in
the root directory.

dependabot[bot] committed:
Bump python from 3.9.6-alpine to 3.10.0-alpine

Bumps python from 3.9.6-alpine to 3.10.0-alpine.

---
updated-dependencies:
- dependency-name: python
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Update for actions/github-script API changes

Update code to reflect breaking changes in V5 of cisagov/github-script as
mentioned here:
https://github.com/actions/github-script#breaking-changes-in-v5

jsf9k committed:
Merge pull request #74 from cisagov/dependabot/docker/python-3.10.0-alpine

Bump python from 3.9.6-alpine to 3.10.0-alpine

mcdonnnj committed:
Merge pull request #91 from cisagov/improvement/pre-commit_update

Update `pre-commit` Hooks

mcdonnnj committed:
Merge pull request #93 from cisagov/improvement/update_ansible-lint_pre-commit_version

Update the `ansible-lint` Version in the pre-commit Configuration

mcdonnnj committed:
Merge pull request #94 from cisagov/improvement/add_dependabot_configuration_for_terraform

Add `dependabot` Configuration for Terraform

mcdonnnj committed:
Merge pull request #92 from cisagov/improvement/use_action_outputs

Use Outputs of cisagov/setup-env-github-action to Control Installed Program Versions

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Update ansible-lint configuration

Remove the manual exclusion of the `converge.yml` file in the default molecule
scenario. With the changes in cisagov/skeleton-ansible-role#88 this exclusion
is no longer necessary.

mcdonnnj committed:
Update ansible-lint configuration

Instead of excluding molecule configurations for systemd individually we
instead direct ansible-lint to process any of these configurations in
scenario directories as plain yaml files. This mirrors how ansible-lint
handles `molecule.yml` files in scenario directories by default.

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`.

dependabot[bot] committed:
Bump python from 3.10.0-alpine to 3.10.1-alpine

Bumps python from 3.10.0-alpine to 3.10.1-alpine.

---
updated-dependencies:
- dependency-name: python
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Merge pull request #97 from cisagov/maintenance/update_pre-commit_hooks

Update `pre-commit` hooks

mcdonnnj committed:
Merge branch 'develop' into improvement/update_ansible-lint_configuration

mcdonnnj committed:
Merge pull request #95 from cisagov/improvement/update_ansible-lint_configuration

Update `ansible-lint` Configuration

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Merge branch 'develop' into lineage/skeleton

mcdonnnj committed:
Merge pull request #72 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into dependabot/docker/python-3.10.1-alpine

mcdonnnj committed:
Merge pull request #75 from cisagov/dependabot/docker/python-3.10.1-alpine

Bump python from 3.10.0-alpine to 3.10.1-alpine

mcdonnnj committed:
Merge branch 'develop' into dependabot/github_actions/actions/github-script-5

mcdonnnj committed:
Merge pull request #73 from cisagov/dependabot/github_actions/actions/github-script-5

Bump actions/github-script from 4 to 5

jsf9k committed:
Merge remote-tracking branch 'skeleton-docker/develop' into skeletonize

jsf9k committed:
Prefer chevron to pystache

pystache is unsupported, and chevron is a faster drop-in replacement.

jsf9k committed:
Remove more obscure Docker platforms

Not all supported platforms can be built in the six hour time limit
imposed by GitHub Actions, so we remove the two most obscure
platforms.

dav3r commented on docker-compose.yml:
```diff
@@ -0,0 +1,25 @@
+    file: ./src/secrets/database_creds.yml
+
+services:
+  trustymail_reporter:
+    # Run the container normally
+    build:
+      context: .
+      dockerfile: Dockerfile
+    depends_on:
+      - redis
+    image: cisagov/trustymail_reporter
```
You almost snuck this one by us! ;)
```suggestion
  pshtt_reporter:
    # Run the container normally
    build:
      context: .
      dockerfile: Dockerfile
    depends_on:
      - redis
    image: cisagov/pshtt_reporter
```

dav3r commented on src/report/create_all_reports.py:
```diff
@@ -0,0 +1,26 @@
+            bashCommand = (
+                HOME_DIR + "/report/generate_https_scan_report.py " + '"' + row[0] + '"'
+            )
+            # generate_trustymail_report.py isn't written in a way
```
Fix a comment:
```suggestion
            # generate_https_scan_report.py isn't written in a way
```

dav3r reviewed (changes requested):
Thanks for your Herculean effort here!  👍 
However, I noted a few small thangs that you missed (they look like copy/paste issues).

jsf9k committed:
Correct references to trustymail_reporter

I copied a few files over from [cisagov/trustymail_reporter](https://github.com/cisagov/trustymail_reporter) when reskeletonizing, since I had already gone through the reskeletonization process for that similarly-structured repo.  It seems I missed a few things.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

dav3r reviewed (approved):
🚀 👍

jsf9k committed:
Fix bug in format of version.txt file

The bump_version.sh and tag.sh scripts expect the version to be
represented in this way.

jsf9k committed:
Bump version from 1.1.14 to 1.2.0

mcdonnnj commented on Dockerfile:
```diff
@@ -44,50 +69,58 @@ RUN apt-get update -qq \
+#
+# Note that matplotlib.basemap is currently incompatible with
+# matplotlib 3.x.
+RUN pip install --no-cache-dir --upgrade pip setuptools \
```
```suggestion
RUN pip install --no-cache-dir --upgrade pip setuptools wheel \
```

mcdonnnj commented on README.md:
```diff
@@ -12,9 +17,235 @@ second-level domains using data collected via
+| Image:tag | Description |
+|-----------|-------------|
+|`cisagov/pshtt_reporter:1.2.0`| An exact release version. |
+|`cisagov/pshtt_reporter:1.1`| The most recent release matching the major and minor version numbers. |
```
```suggestion
|`cisagov/pshtt_reporter:1.2`| The most recent release matching the major and minor version numbers. |
```

mcdonnnj commented on src/report/create_all_reports.py:
```diff
@@ -0,0 +1,26 @@
+        for row in sorted(csv.reader(agency_csv)):
+            bashCommand = (
+                HOME_DIR + "/report/generate_https_scan_report.py " + '"' + row[0] + '"'
+            )
```
Just a thought.

```suggestion
            bashCommand = f'{HOME_DIR}/report/generate_https_scan_report.py "row[0]"'
```

mcdonnnj commented on src/report/generate_https_scan_report.py:
```diff
@@ -0,0 +1,926 @@
+from docopt import docopt
+
+# intra-project modules
+import graphs
```
I prefer absolute imports for things like this.

```suggestion
from . import graphs
```

jsf9k replied:
@mcdonnnj - This change is rejected at runtime due to this code not being in an actual package.  Can I revert the change, or would prefer that I try to turn the `report` package into a proper Python package?

dav3r replied:
In the interest of getting this out the door, I'd vote to revert and create an issue to make `report` a proper package at a later date.

jsf9k replied:
@mcdonnnj - Please see issue #50 and let me know if this is sufficient for your re-approval.  See also commit fa8f6e3.

mcdonnnj commented on src/report/generate_https_scan_report.py:
```diff
@@ -0,0 +1,926 @@
+        # "/reporting.csv" - is this still needed?
+
+    def __latex_escape(self, to_escape):
+        return "".join([LATEX_ESCAPE_MAP.get(i, i) for i in to_escape])
```
```suggestion
        return "".join(LATEX_ESCAPE_MAP.get(i, i) for i in to_escape)
```

mcdonnnj commented on src/report/create_all_reports.py:
```diff
@@ -0,0 +1,26 @@
+            # generate_https_scan_report.py isn't written in a way
+            # that easily allows it to be run in any other way.  Hence
+            # the nosec.
+            os.system(bashCommand)  # nosec B605
```
Should we look at replacing this? Per [Python documentation](https://docs.python.org/3/library/subprocess.html#replacing-os-system):
> The [`os.system()`](https://docs.python.org/3/library/os.html#os.system) function ignores SIGINT and SIGQUIT signals while the command is running, but the caller must do this separately when using the [`subprocess`](https://docs.python.org/3/library/subprocess.html#module-subprocess) module.

jsf9k replied:
Please see commit 524d807.

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -1,40 +1,50 @@
+# matplotlib.pyplot import below.  This also creates a bunch of flake8
+# E402 errors for the imports that follow because they are not at the
+# top of the file, so I added a noqa comment for each of those lines.
+matplotlib.use("Agg")
```
Considering all of the headaches this results in do you think it would be reasonable to use a [`matplotlibrc` file](https://matplotlib.org/2.0.2/users/customizing.html#the-matplotlibrc-file) instead? I mention it after only reading the [backend documentation](https://matplotlib.org/2.0.2/faq/usage_faq.html#what-is-a-backend) so I may not be aware of reasons why we cannot.

jsf9k replied:
This is probably possible, but I'm leery of making changes like that to this legacy code that predates me.  Also, the point of this PR is to reskeletonize this repo; I'd prefer to see this change in a separate PR.  Would you be OK with an issue?

mcdonnnj replied:
Yeah an issue is totally fine. I just thought it worth mentioning given the hoops being jumped through.

jsf9k replied:
See #47.

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -147,34 +183,52 @@ def plot(self, filename, size=1.0):
         ax.set_ylim([-0.5, 5])
         fig.set_tight_layout(True)
-        plt.savefig(filename + '.pdf')
+        plt.savefig(filename + ".pdf")
```
```suggestion
        plt.savefig(f"{filename}.pdf")
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -247,29 +319,47 @@ def plot(self, filename, size=1.0):
 
         fig.set_tight_layout(True)
-        plt.savefig(filename + '.pdf')
+        plt.savefig(filename + ".pdf")
```
```suggestion
        plt.savefig(f"{filename}.pdf")
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -299,65 +390,93 @@ def plot(self, filename, size=1.0):
             else:
                 tick.set_visible(True)
-                tick.set_label('{}'.format(self.series.index[i]))
+                tick.set_label("{}".format(self.series.index[i]))
```
Is there a reason we're doing it this way? I'm not sure if a `str()` wrapper is even necessary, but this `format()` call seems unnecessary.

jsf9k replied:
Please see commit 6d76a32.

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -377,74 +497,98 @@ def too_close(self, trips):
                 new_text = outer.get_text()  # transfer old label text
             else:
-                new_text = '%s\n(%s)' % (outer.get_text(), inner.get_text())
+                new_text = "{}\n({})".format(outer.get_text(), inner.get_text())
```
```suggestion
                new_text = f"{outer.get_text()}\n({inner.get_text()})"
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -377,74 +497,98 @@ def too_close(self, trips):
                 )
             else:
-                label.set_text('%1.1f%%' % (self.data[i]))
+                label.set_text("%1.1f%%" % (self.data[i]))
```
```suggestion
                label.set_text(f"{self.data[i]:1.1f}%")
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -377,74 +497,98 @@ def too_close(self, trips):
-                    self.data[i],
-                    float(self.data[i]) / total * 100.0)
+                label.set_text(
+                    "{}, {:.0f}%".format(
+                        self.data[i], float(self.data[i]) / total * 100.0
+                    )
```
```suggestion
                    f"{self.data[i]}, {(float(self.data[i]) / total * 100.0):.0f}%"
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -458,82 +602,105 @@ def plot(self, filename, size=1.0):
             self.layout_labels(rights, ax, True)
 
-        plt.savefig(filename + '.pdf')
+        plt.savefig(filename + ".pdf")
```
```suggestion
        plt.savefig(f"{filename}.pdf")
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -458,82 +602,105 @@ def plot(self, filename, size=1.0):
-                                        self.label,
-                                        self.agencyScore)
-        federalLabel = 'Federal %s\n%1.2f' % (self.label, self.federalScore)
+        agencyLabel = "{} {}\n{:1.2f}".format(
+            self.agencyName, self.label, self.agencyScore
+        )
```
```suggestion
        agencyLabel = f"{self.agencyName} {self.label}\n{self.agencyScore:1.2f}"
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -458,82 +602,105 @@ def plot(self, filename, size=1.0):
+        agencyLabel = "{} {}\n{:1.2f}".format(
+            self.agencyName, self.label, self.agencyScore
+        )
+        federalLabel = "Federal {}\n{:1.2f}".format(self.label, self.federalScore)
```
```suggestion
        federalLabel = f"Federal {self.label}\n{self.federalScore:1.2f}"
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -458,82 +602,105 @@ def plot(self, filename, size=1.0):
         )
         fig.set_tight_layout(True)
-        plt.savefig(filename + '.pdf')
+        plt.savefig(filename + ".pdf")
```
```suggestion
        plt.savefig(f"{filename}.pdf")
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -548,97 +715,113 @@ def plot(self, filename, size=1.0, figsize=None):
         ax.set_ylim(ymin=0)
         fig.set_tight_layout(True)
-        plt.savefig(filename + '.pdf')
+        plt.savefig(filename + ".pdf")
```
```suggestion
        plt.savefig(f"{filename}.pdf")
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -548,97 +715,113 @@ def plot(self, filename, size=1.0, figsize=None):
         # does work with plt.tight_layout(), but generates a warning
         fig.subplots_adjust(hspace=0)
-        plt.savefig(filename + '.pdf', bbox_inches='tight', pad_inches=0.25)
+        plt.savefig(filename + ".pdf", bbox_inches="tight", pad_inches=0.25)
```
```suggestion
        plt.savefig(f"{filename}.pdf", bbox_inches="tight", pad_inches=0.25)
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -647,47 +830,61 @@ def __init__(self, data_frame, yscale='linear', xlabel=None, ylabel=None,
             tick.label.set_fontsize(6)
         fig.set_tight_layout(True)
-        plt.savefig(filename + '.pdf')
+        plt.savefig(filename + ".pdf")
```
```suggestion
        plt.savefig(f"{filename}.pdf")
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -747,21 +942,24 @@ def plot(self, filename, size=1.0):
         ax.add_collection(patches)
         fig.set_tight_layout(True)
-        plt.savefig(filename + '.pdf')
+        plt.savefig(filename + ".pdf")
```
```suggestion
        plt.savefig(f"{filename}.pdf")
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -799,35 +1013,46 @@ def plot(self, filename, size=1.0):
 
         fig.set_tight_layout(True)
-        plt.savefig(filename + '.pdf')
+        plt.savefig(filename + ".pdf")
```
```suggestion
        plt.savefig(f"{filename}.pdf")
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -843,99 +1068,124 @@ def plot(self, filename, size=1.0):
 
         fig.set_tight_layout(True)
-        plt.savefig(filename + '.pdf')
+        plt.savefig(filename + ".pdf")
```
```suggestion
        plt.savefig(f"{filename}.pdf")
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -843,99 +1068,124 @@ def plot(self, filename, size=1.0):
-        plt.tight_layout()  # trims margins down nicely
-        plt.savefig(filename + '.pdf')  # bbox_inches=0, pad_inches=0
+        plt.tight_layout(rect=[0, 0, 1, 0.98])  # trims margins down nicely
+        plt.savefig(filename + ".pdf")  # bbox_inches=0, pad_inches=0
```
```suggestion
        plt.savefig(f"{filename}.pdf")  # bbox_inches=0, pad_inches=0
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -843,99 +1068,124 @@ def plot(self, filename, size=1.0):
         # plt.show()
         plt.tight_layout()  # trims margins down nicely
-        plt.savefig(filename + '.pdf', bbox_inches=extent, pad_inches=0)
+        plt.savefig(filename + ".pdf", bbox_inches=extent, pad_inches=0)
```
```suggestion
        plt.savefig(f"{filename}.pdf", bbox_inches=extent, pad_inches=0)
```

mcdonnnj commented on tests/container_test.py:
```diff
@@ -0,0 +1,101 @@
+
+# import pytest
+
+ENV_VAR = "ECHO_MESSAGE"
+ENV_VAR_VAL = "Hello World from docker-compose!"
+READY_MESSAGE = "This is a debug message"
+SECRET_QUOTE = (
+    "There are no secrets better kept than the secrets everybody guesses."  # nosec
+)
```
Should we just remove these since they don't seem like they'd be used for testing this container configuration anyway?

```suggestion
```

mcdonnnj reviewed (changes requested):
Thanks for putting in the work on this beast. I had a bunch of syntactic recommendations you're welcome to take or leave for the most part. I also had a few questions that I'd like at least a yay+issue/nay on before proceeding.

jsf9k committed:
Add installation of wheel pip package

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Correct version number in README

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Use f-strings

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Use an absolute import

Also fix imports via isort.

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Prefer a generator to a list comprehension

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Remove unnecessary code

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Remove needless str.format() call

jsf9k committed:
Prefer subprocess to os.system

dav3r reviewed (approved):
Approval intensifies

mcdonnnj commented on tests/conftest.py:
```diff
@@ -0,0 +1,45 @@
+# Third-Party Libraries
+import pytest
+
+MAIN_SERVICE_NAME = "example"
```
I know we're not testing (yet), but for consistency with the `docker-compose.yml`.

```suggestion
MAIN_SERVICE_NAME = "pshtt_reporter"
```

mcdonnnj commented on tests/container_test.py:
```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env pytest -vs
+"""Tests for example container."""
```
```suggestion
"""Tests for pshtt_reporter container."""
```

mcdonnnj commented on src/report.sh:
```diff
@@ -9,9 +9,8 @@ cp ./fonts/* /usr/share/fonts/truetype/
 fc-cache -f
 
 echo 'Waiting for saver'
```
Small nitpick but is there a reason there's usage of both single and double quotes for echo statements of plain text?

jsf9k replied:
Please see commit 5a98f45.

mcdonnnj commented on src/report/create_all_reports.py:
```diff
@@ -0,0 +1,29 @@
+import subprocess  # nosec B404
+
+HOME_DIR = "/home/cisa"
+SHARED_DATA_DIR = HOME_DIR + "/shared/"
```
I should have suggested this the first review just so we're using f-strings consistently.

```suggestion
SHARED_DATA_DIR = f"{HOME_DIR}/shared"
```

mcdonnnj commented on src/report/create_all_reports.py:
```diff
@@ -0,0 +1,29 @@
+
+def main():
+    """Generate all reports."""
+    with open(SHARED_DATA_DIR + "artifacts/unique-agencies.csv") as agency_csv:
```
Together with the above change.

```suggestion
    with open(f"{SHARED_DATA_DIR}/artifacts/unique-agencies.csv") as agency_csv:
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -843,99 +1064,124 @@ def plot(self, filename, size=1.0):
+            plt.text(
+                bar.get_x() + bar.get_width() / 2.0,
+                1.0 * height,
+                "%d" % int(round(height, 0)) + "%",
```
```suggestion
                f"{round(height)}%",
```

mcdonnnj commented on src/report/generate_https_scan_report.py:
```diff
@@ -0,0 +1,926 @@
+PDF_CAPTURE_JS = "pdf_capture.js"
+
+
+class ReportGenerator(object):
```
```suggestion
class ReportGenerator:
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -74,68 +86,92 @@ def wrapLabels(labels, width):
 
 
 class MyMessage(object):
```
```suggestion
class MyMessage:
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -74,68 +86,92 @@ def wrapLabels(labels, width):
         plt.close()
 
 
 class MyStackedBar(object):
```
```suggestion
class MyStackedBar:
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -147,34 +183,52 @@ def plot(self, filename, size=1.0):
         plt.close()
 
 
 class MyBar(object):
```
```suggestion
class MyBar:
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -247,29 +319,47 @@ def plot(self, filename, size=1.0):
         plt.close()
 
 
 class MyDistributionBar(object):
```
```suggestion
class MyDistributionBar:
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -299,65 +390,93 @@ def plot(self, filename, size=1.0):
         plt.close()
 
 
 class MyPie(object):
```
```suggestion
class MyPie:
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -458,82 +600,103 @@ def plot(self, filename, size=1.0):
         plt.close()
 
 
 class MyColorBar(object):
```
```suggestion
class MyColorBar:
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -458,82 +600,103 @@ def plot(self, filename, size=1.0):
         plt.close()
 
 
 class MyLine(object):
```
```suggestion
class MyLine:
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -548,97 +711,113 @@ def plot(self, filename, size=1.0, figsize=None):
         plt.close()
 
 
 class MyPentaLine(object):
```
```suggestion
class MyPentaLine:
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -548,97 +711,113 @@ def plot(self, filename, size=1.0, figsize=None):
         plt.close()
 
 
 class MyStackedLine(object):
```
```suggestion
class MyStackedLine:
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -647,47 +826,61 @@ def __init__(self, data_frame, yscale='linear', xlabel=None, ylabel=None,
         plt.close()
 
 
 class Boxes(object):
```
```suggestion
class Boxes:
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -747,21 +938,24 @@ def plot(self, filename, size=1.0):
         return self.cols
 
 
 class Histogram(object):
```
```suggestion
class Histogram:
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -799,35 +1009,46 @@ def plot(self, filename, size=1.0):
         plt.close()
 
 
 class Histogram2(object):
```
```suggestion
class Histogram2:
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -843,99 +1064,124 @@ def plot(self, filename, size=1.0):
         plt.close()
 
 
 class MyTrustyBar(object):
```
```suggestion
class MyTrustyBar:
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -843,99 +1064,124 @@ def plot(self, filename, size=1.0):
         plt.close()
 
 
 class MyDonutPie(object):
```
```suggestion
class MyDonutPie:
```

mcdonnnj commented on src/report/generate_https_scan_report.py:
```diff
@@ -0,0 +1,926 @@
+            print(
+                'ERROR: "{}" has no live domains - exiting without generating report!'.format(
+                    self.__agency
+                )
```
```suggestion
                f'ERROR: "{self.__agency}" has no live domains - exiting without generating report!'
```

mcdonnnj commented on src/report/generate_https_scan_report.py:
```diff
@@ -0,0 +1,926 @@
+            if self.__agency_id is not None:
+                dest_filename = "{}/cyhy-{}-{}-https-report.pdf".format(
+                    dest_dir, self.__agency_id, datestamp
+                )
```
```suggestion
                dest_filename = f"{dest_dir}/cyhy-{self.__agency_id}-{datestamp}-https-report.pdf"
```

mcdonnnj commented on src/report/generate_https_scan_report.py:
```diff
@@ -0,0 +1,926 @@
+            else:
+                dest_filename = "{}/cyhy-{}-{}-https-report.pdf".format(
+                    dest_dir, self.__agency, datestamp
+                )
```
```suggestion
                dest_filename = f"{dest_dir}/cyhy-{self.__agency}-{datestamp}-https-report.pdf"
```

mcdonnnj commented on src/report/generate_https_scan_report.py:
```diff
@@ -0,0 +1,926 @@
+                        weak_crypto_list.append(wc_text)
+                result = "{}:{} [supports: {}]".format(
+                    hostname, port, ",".join(weak_crypto_list)
+                )
```
```suggestion
                result = f"{hostname}:{port} [supports: {','.join(weak_crypto_list)}]"
```

mcdonnnj commented on src/report/generate_https_scan_report.py:
```diff
@@ -0,0 +1,926 @@
+        xelatex = ["/usr/bin/xelatex", REPORT_TEX]
+        # Bandit frowns upon the use of subprocess, but we need it
+        # here.  Hence the nosec.
+        subprocess.run(xelatex)  # nosec B603
+        subprocess.run(xelatex)  # nosec B603
```
If we know the reason can we get a comment about why this is being done twice?

jsf9k replied:
Please see commit 3256653.

mcdonnnj commented on src/report/generate_https_scan_report.py:
```diff
@@ -0,0 +1,926 @@
+"""Create Cyber Hygiene HTTPS Report PDF.
+
+Usage:
+  generate_https_scan_report [options] "AGENCY"
```
Would it be too much trouble to update this like below? I know it would require updating all references, but I'm really not a fan because it requires accessing the element as `['"AGENCY"']` which is both awkward and serves no benefit.

```suggestion
  generate_https_scan_report [options] AGENCY
```

mcdonnnj commented on src/report/generate_https_scan_report.py:
```diff
@@ -0,0 +1,926 @@
+    db = db_from_config(DB_CONFIG_FILE)
+
+    print("Generating HTTPS Report for {}...".format(args['"AGENCY"']))
+    generator = ReportGenerator(db, args['"AGENCY"'], debug=args["--debug"])
```
This updates for my comment about the Docopt specification. I'm not sure there are any other references.

```suggestion
    print(f"Generating HTTPS Report for {args['AGENCY']}...")
    generator = ReportGenerator(db, args["AGENCY"], debug=args["--debug"])
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -147,34 +183,52 @@ def plot(self, filename, size=1.0):
                 width = int(rect.get_width())
 
-                labelString = '{:,d}'.format(width)
+                labelString = "{:,d}".format(width)
```
```suggestion
                labelString = f"{width:,}"
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -299,65 +390,93 @@ def plot(self, filename, size=1.0):
-
-        plt.bar(pos, self.series.values, tick_label=tick_labels,
-                align='center', color='#000000', edgecolor='#000000')
+            tick_labels[-1] = "{}+".format(tick_labels[-1])
```
```suggestion
            tick_labels[-1] = f"{tick_labels[-1]}+"
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -299,65 +390,93 @@ def plot(self, filename, size=1.0):
-                        ha='right', va='top')
+            plt.axvline(x=day, color="#777777", linewidth=0.5)
+            ax.annotate(
+                "{} Days ".format(day),
```
```suggestion
                "{day} Days ",
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -647,47 +826,61 @@ def __init__(self, data_frame, yscale='linear', xlabel=None, ylabel=None,
+        )
         # set the alpha value of the legend: it will be translucent
         leg.get_frame().set_alpha(0.5)
         for i, tick in enumerate(axes.xaxis.get_major_ticks()):
```
```suggestion
        for _, tick in enumerate(axes.xaxis.get_major_ticks()):
```

mcdonnnj commented on src/report/graphs.py:
```diff
@@ -726,10 +920,7 @@ def plot(self, filename, size=1.0):
             # iterate remainding values of row
             for tup_i, count in enumerate(tup):
                 for k in range(count):
```
```suggestion
                for _ in range(count):
```

mcdonnnj reviewed (approved):
Another slew of Python 3 changes for the most part: Using f-strings and removing `object` from class descriptions. I did have a question about a repeated line of code and a request about the docopt definition for the `generate_https_scan_report.py` file. Thanks again for putting up with all of my nitpicky requests.

jsf9k committed:
Use f-strings

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Drop explicit mention of object for class inheritance

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Use underscores for unused variables

This avoids pollution of scope.

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Update references to "example" in test code

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Add a comment explaining why xelatex is run twice

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Remove needless quotes in echo statements

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Clean up docopt code

Also prefer f-strings to string.format().

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Correct some path references

This commit fixes a few paths that are still referring to the old
/home/reporter path.  Such references should instead be referring to
/home/cisa.

jsf9k committed:
Use Collection.count_documents() instead of Cursor.count()

Cursor.count() has been deprecated for some time, and is now
disallowed.

dav3r reviewed (approved):
Excellent work! 💪

jsf9k commented:
I have confirmed via a full BOD 18-01 reporting run that this code (with the reversion described [here](https://github.com/cisagov/pshtt_reporter/pull/46#discussion_r812292797)) functions as expected.

jsf9k committed:
Revert "Use an absolute import"

This reverts commit 61cf6600d6d37befb1a3569a35827cb080d66d37.

This change was rejected at runtime since the code in src/report does
not constitute a Python package.  See also issue #50.

mcdonnnj commented on Dockerfile:
```diff
@@ -44,50 +69,58 @@ RUN apt-get update -qq \
+###
 
 # Put this just before we change users because the copy (and every
 # step after it) will always be rerun by docker, but we need to be
```
```suggestion
# step after it) will always be rerun by Docker, but we need to be
```

mcdonnnj commented on README.md:
```diff
@@ -12,9 +17,235 @@ second-level domains using data collected via
 This Docker container is intended to be run via
 [orchestrator](https://github.com/cisagov/orchestrator).
```
```suggestion
[cisagov/orchestrator](https://github.com/cisagov/orchestrator).
```

mcdonnnj commented on src/report/generate_https_scan_report.py:
```diff
@@ -0,0 +1,928 @@
+
+# constants
+HOME_DIR = "/home/cisa"
+SHARED_DATA_DIR = HOME_DIR + "/shared/"
```
```suggestion
SHARED_DATA_DIR = f"{HOME_DIR}/shared"
```

mcdonnnj commented on src/report/generate_https_scan_report.py:
```diff
@@ -0,0 +1,928 @@
+SHARED_DATA_DIR = HOME_DIR + "/shared/"
+DB_CONFIG_FILE = "/run/secrets/scan_read_creds.yml"
+HTTPS_RESULTS_CSV_FILE = "pshtt-results.csv"
+OCSP_EXCLUSION_CSV_FILE = SHARED_DATA_DIR + "artifacts/ocsp-crl.csv"
```
```suggestion
OCSP_EXCLUSION_CSV_FILE = f"{SHARED_DATA_DIR}/artifacts/ocsp-crl.csv"
```

mcdonnnj reviewed (approved):
Just some last little suggestions. Really strong work getting this into a more modern state!

jsf9k committed:
Correct capitalization in comment

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Include GitHub org in link text

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Prefer f-strings

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>
